### ISSUE TITLE
Installing Chef 12 on RHEL7

### ISSUE THREAD

**Issue by jimmycliff** [chef/chef#3642]
Hi,

I'm getting a RuntimeError trying to install Chef 12 on RHEL 7.  

I am following the installation instructions at http://docs.chef.io/server/install_server.html#standalone

It appears that the the RHEL7 platform is not rhel?

Screenshot below:

```
[root@cp-chef12dev ~]# rpm -ivh https://web-dl.packagecloud.io/chef/stable/packages/el/7/chef-server-core-12.1.0-1.el7.x86_64.rpm
Retrieving https://web-dl.packagecloud.io/chef/stable/packages/el/7/chef-server-core-12.1.0-1.el7.x86_64.rpm
warning: /var/tmp/rpm-tmp.t46bDz: Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY
Preparing...                          ################################# [100%]
Updating / installing...
   1:chef-server-core-12.1.0-1.el7    ################################# [100%]
[root@cp-chef12dev ~]#
[root@cp-chef12dev ~]# chef-server-ctl install opscode-manage
Starting Chef Client, version 12.4.0.rc.2
resolving cookbooks for run list: ["private-chef::add_ons_wrapper"]
Synchronizing Cookbooks:
  - enterprise
  - yum
  - apt
  - runit
  - private-chef
  - packagecloud
Compiling Cookbooks...

================================================================================
Recipe Compile Error in /opt/opscode/embedded/cookbooks/cache/cookbooks/private-chef/recipes/add_ons_wrapper.rb
================================================================================

RuntimeError
------------
I don't know how to install addons for platform family:

Cookbook Trace:
---------------
  /opt/opscode/embedded/cookbooks/cache/cookbooks/private-chef/recipes/add_ons_remote.rb:44:in `from_file'
  /opt/opscode/embedded/cookbooks/cache/cookbooks/private-chef/recipes/add_ons_wrapper.rb:47:in `from_file'

Relevant File Content:
----------------------
/opt/opscode/embedded/cookbooks/cache/cookbooks/private-chef/recipes/add_ons_remote.rb:

 37:      gpgcheck true
 38:      action :create
 39:    end
 40:
 41:  else
 42:    # TODO: probably don't actually want to fail out?  Say, on any platform where
 43:    # this would have to be done manually.
 44>>   raise "I don't know how to install addons for platform family: #{node['platform_family']}"
 45:  end
 46:
 47:  node['private_chef']['addons']['packages'].each do |pkg|
 48:    package pkg do
 49:      notifies :create, "ruby_block[addon_install_notification_#{pkg}]", :immediate
 50:    end
 51:  end
 52:


Running handlers:
  - #<Class:0x00000003abf2b8>::AddonInstallHandler
Running handlers complete
Chef Client failed. 0 resources updated in 1.817571381 seconds
[2015-07-07T14:59:27-04:00] FATAL: Stacktrace dumped to /opt/opscode/embedded/cookbooks/cache/chef-stacktrace.out
[2015-07-07T14:59:27-04:00] FATAL: Chef::Exceptions::RunFailedWrappingError: Found 1 errors, they are stored in the backtrace
[root@cp-chef12dev ~]# more /opt/opscode/embedded/cookbooks/cache/chef-stacktrace.out
Generated at 2015-07-07 14:59:27 -0400
Chef::Exceptions::RunFailedWrappingError: Found 1 errors, they are stored in the backtrace
1) RuntimeError -  I don't know how to install addons for platform family:
/opt/opscode/embedded/cookbooks/cache/cookbooks/private-chef/recipes/add_ons_remote.rb:44:in `from_file'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/mixin/from_file.rb:30:in `instance_eval'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/mixin/from_file.rb:30:in `from_file'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/cookbook_version.rb:228:in `load_recipe'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context.rb:174:in `load_recipe'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context.rb:143:in `block in include_recipe'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context.rb:142:in `each'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context.rb:142:in `include_recipe'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/dsl/include_recipe.rb:26:in `include_recipe'
/opt/opscode/embedded/cookbooks/cache/cookbooks/private-chef/recipes/add_ons_wrapper.rb:47:in `from_file'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/mixin/from_file.rb:30:in `instance_eval'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/mixin/from_file.rb:30:in `from_file'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/cookbook_version.rb:228:in `load_recipe'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context.rb:174:in `load_recipe'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context/cookbook_compiler.rb:140:in `block in compile_recipes'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context/cookbook_compiler.rb:138:in `each'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context/cookbook_compiler.rb:138:in `compile_recipes'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context/cookbook_compiler.rb:75:in `compile'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/run_context.rb:97:in `load'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/policy_builder/expand_node_object.rb:87:in `setup_run_context'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/client.rb:480:in `setup_run_context'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/client.rb:266:in `run'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/application.rb:252:in `run_with_graceful_exit_option'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/application.rb:229:in `block in run_chef_client'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/local_mode.rb:39:in `with_server_connectivity'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/application.rb:212:in `run_chef_client'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/application/client.rb:375:in `run_application'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/lib/chef/application.rb:60:in `run'
/opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.4.0.rc.2/bin/chef-client:26:in `<top (required)>'
/opt/opscode/embedded/bin/chef-client:23:in `load'
/opt/opscode/embedded/bin/chef-client:23:in `<main>'

```

**Comment by coderanger** [comment]
Chef Server is not supported on RHEL 7 at this time. Additionally this would be an issue for the chef-server project.

**Comment by jimmycliff** [comment]
Ok thank you.

**Comment by sdelano** [comment]
Hi @jimmycliff - I've copied this ticket over to the chef-server repo (as shown above in the mention). Chef Server is built and tested on RHEL7, so it should work just fine there. Where you're running into this issue is actually installing the Chef Manage addon. You should be able to work around this by directly downloading the opscode-manage package from https://downloads.chef.io.

It appears that this is due to chef-client not correctly recognizing the platform_family attribute on RHEL7. According to @jaym, this issue should be fixed in Chef Client 12.4.1.

**Comment by sdelano** [comment]
To clarify - this has been fixed in master of ohai and will be fixed in the next release of Chef Server.

[Ohai commit](https://github.com/chef/ohai/commit/eccaff433d062a9d2ece196480032b7b96bf942a)
